ale/drivers/selene_drivers.py

```diff
@@ -520,33 +549,302 @@ def detector_start_sample(self):
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector lines
+        """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, 1/pixel_size, 0]
+
+    @property
+    def focal2pixel_samples(self):
+        """
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector samples
+        """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, 0, -1/pixel_size]
+
+    @property
+    def detector_center_line(self):
+        """
+        Returns the center detector line of the detector. Expects ikid to be
+        defined. This should be the NAIF integer ID code for the sensor.
+
+        We subtract 0.5 from the center line because as per the IK:
+        Center of the first pixel is defined as "1.0".
+
         Returns
         -------
         : int
-          ISIS sensor model version
+          The detector line of the principle point
+        """
+        return spice.gdpool('INS{}_BORESIGHT_LINE'.format(self.ikid), 0, 1)[0]
+
+    @property
+    def detector_center_sample(self):
+        """
+        Returns the center detector sample of the detector. Expects ikid to be
+        defined. This should be the NAIF integer ID code for the sensor.
+
+        We subtract 0.5 from the center sample because as per the IK:
+        Center of the first pixel is defined as "1.0".
+
+        Returns
+        -------
+        : int
+          The detector sample of the principle point
+        """
+        return spice.gdpool('INS{}_BORESIGHT_SAMPLE'.format(self.ikid), 0, 1)[0]
+
+    @property
+    def _odkx(self):
+        """
+        Returns the x coefficients of the optical distortion model.
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          Optical distortion x coefficients
         """
+        return spice.gdpool('INS{}_DISTORTION_COEF_X'.format(self.ikid),0, 4).tolist()
+
+    @property
+    def _odky(self):
+        """
+        Returns the y coefficients of the optical distortion model.
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          Optical distortion y coefficients
+        """
+        return spice.gdpool('INS{}_DISTORTION_COEF_Y'.format(self.ikid), 0, 4).tolist()
+
+    @property
+    def boresight_x(self):
+        """
+        Returns the x focal plane coordinate of the boresight.
+        Expects ikid to be defined. This should be the NAIF integer ID for the
```

Comment:
Do we needs these expects? The API called doesn't need to know this information because this class has an ikid property. The code is explicit that self.ikid is needed. Maybe these 'expects` in the docstring can be removed?

Reply:
They were/are a way to tell people what other properties a property relies on, since many of the properties have default implementations

Reply:
I see that, but this is a 3 line function and the code clearly needs the ikid.